### PR TITLE
handle multiple graphql clients by providing a schema

### PR DESCRIPTION
**Pull request by craigiek**

### CODE REVIEW

File: lib/buildr_plus/features/graphql_client.rb [hunk 1]
```diff
@@ -15,12 +15,20 @@
...
+      self.graphql_schemas[graphql_schema_name] = graphql_schema_artifact
     end
 
-    attr_reader :graphql_schema_name
-    attr_reader :graphql_schema_artifact
+    def graphql_schemas
+      @graphql_schemas ||= {}
+    end
+
+    def graphql_schema_name(name = :default)
```
Comment (realityforge) [inline lib/buildr_plus/features/graphql_client.rb:25]: As this just returns the name parameter we can probably ditch this method

File: lib/buildr_plus/features/graphql_client.rb [hunk 2]
```diff
@@ -15,12 +15,20 @@
...
+      self.graphql_schemas[name]
+    end
+
+    def graphql_schema_artifact(name = :default)
```
Comment (realityforge) [inline lib/buildr_plus/features/graphql_client.rb:29]: We can probably ditch ` = :default` as it seems we always pass in the name?

File: lib/buildr_plus/features/graphql_client.rb [hunk 3]
```diff
@@ -15,12 +15,20 @@
...
+      self.graphql_schemas[name]
+    end
 
     def generate
```
Comment (realityforge) [inline lib/buildr_plus/features/graphql_client.rb:33]: This has not been updated? We either need to make sure intellij can gracefully handle it's absence or fix it

File: lib/buildr_plus/features/graphql_client.rb (hunk 2 above)
Comment (craigiek) [reply lib/buildr_plus/features/graphql_client.rb:29]: Not sure about this one - look further down the file lines 51 and 54

**Pull request merged**